cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -79,20 +83,26 @@ public Path getBasePath() {
   public static void validateConfig(SecureStore secureStore, SourceControlConfig sourceControlConfig) throws
     RepositoryConfigValidationException {
     RepositoryConfig config = sourceControlConfig.getRepositoryConfig();
+    CredentialsProvider credentialsProvider = sourceControlConfig.getAuthenticationStrategy()
+      .getCredentialsProvider(secureStore, config, sourceControlConfig.getNamespaceID());
     // Try fetching heads in the remote repository.
     try (Git git = Git.wrap(new InMemoryRepository.Builder().build())) {
-      LsRemoteCommand cmd = createCommand(git::lsRemote, sourceControlConfig, secureStore).setRemote(config.getLink())
-                                                                                          .setHeads(true)
-                                                                                          .setTags(false);
+      LsRemoteCommand cmd =
+        createCommand(git::lsRemote, sourceControlConfig, credentialsProvider).setRemote(config.getLink())
+          .setHeads(true)
+          .setTags(false);
       validateDefaultBranch(cmd.callAsMap(), config.getDefaultBranch());
     } catch (TransportException e) {
       throw new RepositoryConfigValidationException(
         "Failed to connect with repository. Authentication credentials invalid or the repository may not exist.",
         e);
     } catch (RepositoryConfigValidationException e) {
       throw e;
-    } catch (AuthenticationConfigException e) {
+    } catch (IllegalStateException e) {
       throw new RepositoryConfigValidationException(e.getMessage(), e);
+    } catch (GitAPIException e) {
+      throw new RepositoryConfigValidationException("Failed to list remotes in remote repository: " + e.getMessage(),
```

Comment:
why have different handling for IllegalState, this, and the generic Exception?

Reply:
IllegalState was there to catch exceptions thrown by CredentialsProvider when there was no refresh() method, its no longer caught. 
For GitAPIException I'm adding the exception message in the RepositoryConfigValidationException so that it's visible to the user.
For any other Exception, the message of the inner exception isn't added as it may contain internal details of the code that the user may not understand.